### PR TITLE
Make it easier for devs to add CalciteQueryTests

### PR DESCRIPTION
**Pull request by suneet-s**
Writing a CalciteQueryTest can involve a lot of mental overhead. This discourages devs from
adding to these tests. Some examples:
 * What should the native query look like for my sql?
 * What does the schema of the different tables that CalciteQueryTest use look like? Which columns contain nulls, multi-values, etc.
 * What does the lookup look like?

Because of this, when some users see sql issues, they are not able to provide a test that shows the issue.

This PR aims to make it easier for both devs and users to add to the CalciteQueryTests by
 - Providing ingestion specs so that a user can easily recreate the CalciteQueryTests in their local druid setup.
 - Adding a json spec where queries and expected results can be added to be tested.

### CODE REVIEW

**Comment by jihoonson** [comment]
@suneet-s would you elaborate more on what the purpose of this PR is? Do you just want to keep them somewhere so that you can use locally? I think how to manage them should be answered. My main questions are:

1) Should data be synced between CalciteTests and ingestion specs? 
2) Also, directly setting rows and using ingestion specs can result in different data stored in segments, for example when rollup is set. How do we manage this? 
3) Finally, since ingestion specs don't look related to any other codes directly, it's likely possible that someone will delete those ingestion specs in the future. How can we avoid this?

**Comment by suneet-s** [comment]
> @suneet-s would you elaborate more on what the purpose of this PR is? Do you just want to keep them somewhere so that you can use locally? I think how to manage them should be answered. My main questions are:

Adding CalciteQueryTests was quite painful for me because I did not have the ability to easily look at the datasources to see what was in them. I added these ingestion specs for my local cluster to make testing and writing SQL easier. Before this change, I'd have to test against a wikipedia datasource and then figure out how to translate that query to one of the datasources in the CalciteQueryTest. For now, I was hoping they could live here to help future devs with writing tests.

> 1. Should data be synced between CalciteTests and ingestion specs?

I don't think they need to be synced. CalciteTests are based on the data specified in this class. The ingestion specs are merely helpers

> 2. Also, directly setting rows and using ingestion specs can result in different data stored in segments, for example when rollup is set. How do we manage this?

I believe all the ingestion specs specify rollup is disabled so that they mimic what the tables look like in CalciteTests

> 3. Finally, since ingestion specs don't look related to any other codes directly, it's likely possible that someone will delete those ingestion specs in the future. How can we avoid this?

Good question. I was deciding between leaving these in the druid-sql package vs the druid-integration-tests package. Eventually I want to move these to the integration testing framework. This way devs can write easy sql validation integration tests without having to worry about how Calcite translates the query to a native query. I was planning on moving it into the integration testing framework as part of another patch. Hopefully that patch will make it a lot easier for any dev to write a SQL integration test against these datasources.

**Comment by jihoonson** [comment]
I think now I see more clear what you want to do. This change doesn't seem doing anything with CalciteQueryTests, but is more for local testing and potentially integration testing in the future. If it's the case, the title seems confusing to me. 

> > 2. Also, directly setting rows and using ingestion specs can result in different data stored in segments, for example when rollup is set. How do we manage this?
> 
> I believe all the ingestion specs specify rollup is disabled so that they mimic what the tables look like in CalciteTests

Sorry, I meant the ingestion specs can be evolved independently from the data in CalciteTests. I think it would be fine if they don't have to be synced.

> > 3. Finally, since ingestion specs don't look related to any other codes directly, it's likely possible that someone will delete those ingestion specs in the future. How can we avoid this?
> 
> Good question. I was deciding between leaving these in the druid-sql package vs the druid-integration-tests package. Eventually I want to move these to the integration testing framework. This way devs can write easy sql validation integration tests without having to worry about how Calcite translates the query to a native query. I was planning on moving it into the integration testing framework as part of another patch. Hopefully that patch will make it a lot easier for any dev to write a SQL integration test against these datasources.

Why do we want to not worry about Calcite planning? What the SQL layer does in Druid is mostly query planning. It doesn't seem right to me to not test query planning part for SQL layer. Or am I missing something?

This makes me curious what kind of integration testing we want for SQL. Per my understanding, the integration testing should verify the integrity of the entire Druid system. AFAIT, the missing parts in SQL layer are 1) querying SQL via HTTP or JDBC, and 2) querying against large data (this is more like for query processing engine rather than SQL but would be nice to run using SQL since it's easier). The details of query planning, query processing, etc should be covered by unit testing because integration testing takes too long. For 1), we don't need to test various queries since it focuses on `via HTTP` and `via JDBC` parts. For 2), we need to test various queries but also need large data. For integration test data, we are currently using precreated segments as well as synthetic data generator. The new ingestion specs doesn't seem to have benefits over existing data in neither cases. It might be more useful to share the queries in CalciteQueryTest and 2).

**Comment by ccaominh** [comment]
What do you think about adding a README to ` sql/src/test/resources/calcite/tests` to help explain the intent of the json files and @jihoonson's suggestion of rewording the PR title/description?

File: sql/src/test/resources/calcite/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+This package contains ingestion specs for datasources used by Calcite Tests.
+
+The purpose of these files is to make it easier to look at and manipulate the data under test so that you can easily
+validate if the results of the SQL query written are as expected. 
+
+> NOTE: Thes provided specs are not guaranteed to be in sync with the datasources used by the test. The source of truth
```
Comment (ccaominh) [inline sql/src/test/resources/calcite/tests/README.md:6]: There's a typo: "NOTE: Thes"

Comment (suneet-s) [reply sql/src/test/resources/calcite/tests/README.md:6]: 🤦 
```suggestion
> NOTE: The provided specs are not guaranteed to be in sync with the datasources used by the test. The source of truth
```

**Pull request merged**